snuba/snuba_migrations/querylog/0004_add_bytes_scanned.py

```diff
@@ -0,0 +1,55 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Array, Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    Adds fields for query profile.
+    """
+
+    blocking = True
+
+    def __forward_migrations(
+        self, table_name: str
+    ) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.QUERYLOG,
+                table_name=table_name,
+                column=Column(
+                    "clickhouse_queries.bytes_scanned",
+                    Array(
+                        UInt(64),
+                        Modifiers(
+                            default="arrayResize([0], length(clickhouse_queries.sql))"
+                        ),
```

Comment:
You should be able to remove this as you are now writing the actual empty array.

Reply:
But not all rows have this column written. Doesn't it need to exist until that's no longer the case?

Reply:
If not all rows have this column written, please hold off, that would break the consumer. 

In a nested column all rows MUST have all columns written. When you write a nested column you write an array per column. The arrays MUST have the same length or Clickhouse will reject the write and the consumer will get stuck.

Reply:
No, because the column does not exist in the DB. 
The critical part is that you deploy the change that writes the column for all rows before we apply the migration. 

The default here proved to be useless right after we added it back in 2020. 
It was a desperate move to make a migration work without having to stop the consumer. We did not know about `input_format_skip_unknown_fields` back then.